elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideDynamicConfiguration.java

```diff
@@ -0,0 +1,148 @@
+            Map<String, Object> puiPropertyMap = new HashMap<>();
+
+            //Bind entity classes from classpath to Persistence Unit
+            ArrayList<Class> bindClasses = new ArrayList<>();
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(Entity.class));
+
+            //Bind FromTable/FromSubSelect classes from classpath to Persistence Unit
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(FromTable.class));
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(FromSubquery.class));
+
+            //Map of JPA Properties to be be passed to EntityManager
+            Map<String, String> jpaPropMap = jpaProperties.getProperties();
+            String hibernateGetDDLAuto = hibernateProperties.getDdlAuto();
+
+            //Set the relevant property in JPA corresponding to Hibernate Property Value
+            if (jpaPropMap.get(HIBERNATE_DDL_AUTO) == null && hibernateGetDDLAuto != null) {
+               jpaPropMap.put(HIBERNATE_DDL_AUTO, hibernateGetDDLAuto);
+             }
+
+            ElideDynamicEntityCompiler compiler = dynamicCompiler.getIfAvailable();
+
+            Collection<ClassLoader> classLoaders = new ArrayList<>();
+            classLoaders.add(compiler.getClassLoader());
+
+            //Add dynamic classes to Pui Map
+            puiPropertyMap.put(AvailableSettings.CLASSLOADERS, classLoaders);
+            //Add classpath entity model classes to Pui Map
+            puiPropertyMap.put(AvailableSettings.LOADED_CLASSES, bindClasses);
+
+            //pui properties from pui map
+            Properties puiProps = new Properties();
+            puiProps.putAll(puiPropertyMap);
+
+            //Create Elide dynamic Persistence Unit
+            ElideDynamicPersistenceUnit elideDynamicPersistenceUnit =
+                    new ElideDynamicPersistenceUnit("dynamic", compiler.classNames, puiProps,
+                    compiler.getClassLoader());
+            elideDynamicPersistenceUnit.setNonJtaDataSource(source);
+            elideDynamicPersistenceUnit.setJtaDataSource(source);
+
+            HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+            vendorAdapter.setShowSql(jpaProperties.isShowSql());
+            vendorAdapter.setGenerateDdl(jpaProperties.isGenerateDdl());
+            if (jpaProperties.getDatabase() != null) {
+                vendorAdapter.setDatabase(jpaProperties.getDatabase());
+            }
+            if (jpaProperties.getDatabasePlatform() != null) {
+                vendorAdapter.setDatabasePlatform(jpaProperties.getDatabasePlatform());
+            }
+
+            LocalContainerEntityManagerFactoryBean bean = new LocalContainerEntityManagerFactoryBean();
+            bean.setJpaVendorAdapter(vendorAdapter);
+
+            //Add JPA Properties from Application.yaml
+            bean.setJpaPropertyMap(jpaPropMap);
+
+            //Add Classes
+            bean.setJpaPropertyMap(puiPropertyMap);
```

Comment:
Does this overwrite the previous or extend it?

Reply:
It extends the jpaPropertyMap to include the classloader and annotated classes